entry_types/scrolled/package/src/frontend/Video.js

```diff
@@ -6,95 +6,70 @@ import MutedContext from './MutedContext';
 
 import styles from './Video.module.css';
 
-export function Video(props) {
-  const awsBucket = '//s3-eu-west-1.amazonaws.com/de.codevise.pageflow.development/pageflow-next/presentation-videos/';
-
-  const videoBoatSunset  = awsBucket+'floodplain-clean.mp4';
-  const poster_videoBoatSunset  = awsBucket+'posterframes/poster_katerchen.jpeg';
-  const videoBoatDark  = awsBucket+'floodplain-dirty.mp4';
-  const poster_videoBoatDark  = awsBucket+'posterframes/poster_katerchen.jpeg';
-  const videoKaterchen  = awsBucket+'katerchen.mp4';
-  const poster_videoKaterchen  = awsBucket+'posterframes/poster_katerchen.jpeg';
-  const videoGarzweilerLoop1  = awsBucket+'braunkohle_loop1.mp4';
-  const poster_videoGarzweilerLoop1  = awsBucket+'posterframes/poster_braunkohle_loop1.jpeg';
-  const videoGarzweilerLoop2  = awsBucket+'braunkohle_loop2.mp4';
-  const poster_videoGarzweilerLoop2  = awsBucket+'posterframes/poster_braunkohle_loop2.jpeg';
-  const videoGarzweilerDrohne  = awsBucket+'braunkohle_drone.mp4';
-  const poster_videoGarzweilerDrohne  = awsBucket+'posterframes/poster_braunkohle_drone.jpeg';
-  const videoInselInterviewToni  = awsBucket+'pageflow_insel_interview_toni02.mp4';
-  const poster_videoInselInterviewToni  = awsBucket+'posterframes/poster_pageflow_insel_interview_toni02.jpg';
+import {useFile} from '../entryState';
 
-  const videoUrl = {
-    videoBoatSunset,
-    videoBoatDark,
-    videoKaterchen,
-    videoGarzweilerLoop1,
-    videoGarzweilerLoop2,
-    videoGarzweilerDrohne,
-    videoInselInterviewToni
-  }[props.id];
-
-  const posterUrl = {
-    poster_videoBoatSunset,
-    poster_videoBoatDark,
-    poster_videoKaterchen,
-    poster_videoGarzweilerLoop1,
-    poster_videoGarzweilerLoop2,
-    poster_videoGarzweilerDrohne,
-    poster_videoInselInterviewToni
-  }['poster_'+props.id];
+/**
+ * Render a video file.
+ *
+ * @param {Object} props
+ * @param {number} props.id - Perma id of the video file.
+ */
+export function Video(props) {
+  const videoFile = useFile({collectionName: 'videoFiles', permaId: props.id});
 
-  const videoRef = useRef();
-  const state = props.state;
+  if (videoFile) {
+    const videoRef = useRef();
```

Comment:
The lint warnings are right. Hooks can not be used conditionally. This also leads to errors in the browser consoles. You could extract a component that uses the hooks instead and render _it_ conditionally.

Reply:
Refactored to component